src/mame/barcrest/mpu4.cpp

```diff
@@ -262,37 +262,36 @@ void mpu4_state::lamp_extend_small(int data)
 		}
 	}
 	m_lamp_strobe_ext_persistence ++;
-	if ((m_lamp_strobe_ext_persistence == 3)||(m_lamp_strobe_ext!=column))
+	if ((m_lamp_strobe_ext_persistence == 3)||(m_lamp_strobe_ext[0] != column))
 	{
 		m_lamp_strobe_ext_persistence = 0;
-		m_lamp_strobe_ext=column;
+		m_lamp_strobe_ext[0] = column;
 	}
 }
 
 void mpu4_state::lamp_extend_large(int data,int column,int active)
 {
-	int lampbase,i,bit7;
-
 	m_lamp_sense = 0;
-	bit7 = data & 0x80;
+	int bit7 = (data & 0x80)>>7;
```

Comment:
You may as well use `BIT(data, 7)` – you’re even calling it `bit7`.  It’s less error-prone that way as you only have one integer literal (the bit position) rather than two that need to be in sync. (the shift and mask).